test/e2e/baseline_test.go

```diff
@@ -909,3 +910,37 @@ func PromDiff(t test.Failer, prom prometheus.Instance, cluster cluster.Cluster,
 
 	}
 }
+
+func TestServiceRestart(t *testing.T) {
+	const callInterval = 100 * time.Millisecond
+	successThreshold := 1.0
+	framework.NewTest(t).Run(func(t framework.TestContext) {
+		for _, dst := range apps.EnrolledToKmesh {
+			generators := []traffic.Generator{}
+			mkGen := func(src echo.Caller) {
+				g := traffic.NewGenerator(t, traffic.Config{
+					Source: src,
+					Options: echo.CallOptions{
+						To:    dst,
+						Count: 1,
+						Check: check.OK(),
+						HTTP:  echo.HTTP{Path: "/?delay=10ms"},
+						Port: echo.Port{
+							Name: "http",
+						},
+						Retry: echo.Retry{NoRetry: true},
+					},
+					Interval: callInterval,
+				}).Start()
+				generators = append(generators, g)
+			}
+			mkGen(dst)
```

Comment:
```suggestion
			apps.ServiceWithWaypointAtServiceGranularity[0]
```

Not use `dst` itself as `src`.